### DISCUSSION TITLE
Upgrade policy for prometheus operator

### DISCUSSION THREAD

**Discussion by saurabhvagrawal** [prometheus-operator/prometheus-operator#3953] [Q&A]
Hi Team,

We are using prometheus operator in our production clusters. We can see that there are frequent releases for prometheus operator. Wanted to understand what is the best approach of upgrading it. Should we update it to version n(latest) or n-1 or n-2 whenever we plan our next upgrade.

Appreciate your response.

**Comment by brancz** [comment]
There are plenty of upgrade scenarios tested in CI, so if that's your worry waiting is unlikely to increase upgrade confidence. Of course, new features may have an impact, but typically new features tend to be new fields, so one could say it's feature gates almost. We at Polar Signals just upgrade to whatever latest version is available in [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus), and that hasn't caused any issues.

**Reply by simonpasquier** [reply]
I'll second Frederic's answer. The operator is part of the monitoring stack shipped by OpenShift and we tend to stick with the latest stable release as much as possible.

**Reply by saurabhvagrawal** [reply]
Thanks for your reply. I agree with all the inputs you guys have provided. I just wanted to understand the industry standard and typically what is followed across the board to get more confidence when we move with this process.

**Comment by mpechner-akasa** [comment]
Not seeing an answer. 
I need to upgrade from ver 42.x to 46.x
Can do it in one step, or multiple:  42-43, 43-44, 44-45, 45-46?

**Comment by data-dude** [comment]
Is there documentation on the best way to update?  I installed with Helm and I'm not comfortable with just updating everything all at once.  If there's an error how do I revert back? and other concerns.  It'd be nice if there was a clear guide.  1) Update crds first?  Then roles?

**Reply by mpechner-akasa** [reply]
I have been updating once every 60days.  Have mnot had a failure yet.
I apply the CRDS, then the operator helm.  
No issues.

the update notes always say to update the CRDs first.

**Reply by data-dude** [reply]
OK Thanks
